packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -0,0 +1,427 @@
+   * The first parameter is the event. The second parameter is the changed onsetDate value.
+   */
+  onsetDateInputOnChange: PropTypes.func,
+};
+
+const defaultProps = {
+  birthdate: undefined,
+  granularity: 'MONTH',
+  granularitySelectName: undefined,
+  granularitySelectOnChange: undefined,
+  precision: 'ABOUT',
+  precisionSelectName: undefined,
+  precisionSelectOnChange: undefined,
+  onsetDate: moment().format('YYYY-MM-DD'),
+  onsetDateInputName: undefined,
+  onsetDateInputOnChange: undefined,
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+class OnsetPicker extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      granularity: this.props.granularity,
+      precision: this.props.precision,
+      onsetDate: this.props.onsetDate,
+      ageCalcCount: 1,
+      ageCalcDuration: 'weeks',
+    };
+
+    // This binding is necessary to make `this` work in the callback
+    this.changeGranularity = this.changeGranularity.bind(this);
+    this.changePrecision = this.changePrecision.bind(this);
+    this.changeAgeCalcCount = this.changeAgeCalcCount.bind(this);
+    this.changeAgeCalcDuration = this.changeAgeCalcDuration.bind(this);
+    this.changeYear = this.changeYear.bind(this);
+    this.changeMonth = this.changeMonth.bind(this);
+    this.changeDate = this.changeDate.bind(this);
+    this.availableMonths = this.availableMonths.bind(this);
+    this.availableYears = this.availableYears.bind(this);
+    this.maxAgeCount = this.maxAgeCount.bind(this);
+    this.allowedAgeDurations = this.allowedAgeDurations.bind(this);
+  }
+
+  /*
+   * Change state for granularity
+   */
+  changeGranularity(event) {
+    this.setState({ granularity: event.target.value });
+
+    if (this.props.granularitySelectOnChange) {
+      this.props.granularitySelectOnChange(event, event.target.value);
+    }
+  }
+
+  /*
+   * Change state for precision
+   */
+  changePrecision(event) {
+    this.setState({ precision: event.target.value });
+
+    if (this.props.precisionSelectOnChange) {
+      this.props.precisionSelectOnChange(event, event.target.value);
+    }
+  }
+
+  /*
+   * Change state for count when using age granularity, and update onset date
+   */
+  changeAgeCalcCount(event) {
+    const newDate = moment(this.props.birthdate).add(Number(event.target.value), this.state.ageCalcDuration).format('YYYY-MM-DD');
+
+    this.setState({
+      ageCalcCount: Number(event.target.value),
+      onsetDate: newDate });
+
+    if (this.props.onsetDateInputOnChange) {
+      this.props.onsetDateInputOnChange(event, newDate);
+    }
+  }
+
+  /*
+   * Change state for duration when using age granularity, and update onset date
+   */
+  changeAgeCalcDuration(event) {
```

Comment:
Should this be called changeOnsetDate?

Reply:
This is for handling the age unit onChange, and has been renamed to changeAgeUnit to reflect that.